From cc804a442827b05baace67522c29e8823a6bafb6 Mon Sep 17 00:00:00 2001
From: Simon Thelen <ffmpeg-dev@c-14.de>
Date: Sat, 8 Apr 2017 14:01:39 +0200
Subject: [PATCH 1/3] libavformat/tcp: fix return code for tcp_accept

ff_accept can return AVERROR(ETIMEDOUT) and errno will be 0 (or
undefined), return ret instead and return ff_neterror() in
ff_poll_interrupt instead of AVERROR(errno) to parse WSAGetLastError on
Windows.
---
 libavformat/network.c | 2 +-
 libavformat/tcp.c     | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/libavformat/network.c b/libavformat/network.c
index 2fb1c8b..b3987a4 100644
--- a/libavformat/network.c
+++ b/libavformat/network.c
@@ -159,7 +159,7 @@ static int ff_poll_interrupt(struct pollfd *p, nfds_t nfds, int timeout,
     if (!ret)
         return AVERROR(ETIMEDOUT);
     if (ret < 0)
-        return AVERROR(errno);
+        return ff_neterrno();
     return ret;
 }
 
diff --git a/libavformat/tcp.c b/libavformat/tcp.c
index 3055e48..07b4ed9 100644
--- a/libavformat/tcp.c
+++ b/libavformat/tcp.c
@@ -204,7 +204,7 @@ static int tcp_accept(URLContext *s, URLContext **c)
     cc = (*c)->priv_data;
     ret = ff_accept(sc->fd, sc->listen_timeout, s);
     if (ret < 0)
-        return ff_neterrno();
+        return ret;
     cc->fd = ret;
     return 0;
 }
-- 
2.9.3

